src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce<GroupsMap<T>>((acc, item) => {
+    const keyValue = item[key] as string;
```

Comment:
The code assumes that the `key` value is always a string. If `key` is not a string, this could lead to runtime errors. Consider using `String(item[key])` to ensure the key value is always a string.